### DISCUSSION TITLE
Jupyter / xeus-cling Instructions

### DISCUSSION THREAD

**Discussion by TwoUnderscorez** [alandefreitas/matplotplusplus#373] [Q&A] [answered]
Are there any instructions on getting this library to work with Jupyter / JupyterLab (with xeus-cling)?
It seems that as of now cling does not support linking with static libraries, only shared objects. 
Any help would be appriciated.

**Comment by alandefreitas** [comment]
I don't think it would easily work with xeux-cling because it has a specific format to plot things in the notebook. There might be an intrusive solution via a backend that would render the plots, save them on a file, read the file back in some format, and convert this format to whatever format xeus-cling requires. But that doesn't exist yet. Ideally, we would have a better backend API where we bypass gnuplot completely, which has always been my intention, but I don't have the resources to work on that at the moment.

**Reply by TwoUnderscorez** [reply]
I managed to compile matplot++ as a shared library and load it, but I can't seem to be able to save the image. Do you have any advice by any chance?
![image](https://github.com/alandefreitas/matplotplusplus/assets/25303006/4e1c569a-63b7-4924-a5c9-639bb7d3ee9b)
(When I run it inside JupyterLab the kernel crashes without any other error messages so I assumed fixing this will help).

**Reply by alandefreitas** [reply, marked as answer]
I think this is because of the Qt gnuplot terminal. I would make sure this terminal works outside cling and then check if other terminals work. I think you would have a look at options to change the terminal or the default terminal in the header files because this is probably not in the examples.

**Reply by TwoUnderscorez** [reply]
I had to set the environment variable `GNUTERM` to `png`, then it worked.
Thanks.
